packages/app-mobile/components/screens/Note.tsx

```diff
@@ -434,8 +530,72 @@ class NoteScreenComponent extends BaseScreenComponent {
 		// has already been granted, it doesn't slow down opening the note. If it hasn't
 		// been granted, the popup will open anyway.
 		void this.requestGeoLocationPermissions();
+
+		this.unsubscribeScreenWidthChangeHandler_ = Dimensions.addEventListener('change', this.handleScreenWidthChange_);
 	}
 
+	private animateNotesBarOpen = () => {
+		Animated.parallel([
+			Animated.spring(
+				this.notesBarPosition,
+				{
+					toValue: 0,
+					duration: 1500,
+				}
+			),
+			Animated.spring(
+				this.notePosition,
+				{
+					toValue: 0,
+					duration: 1500,
+				}
+			),
+			Animated.spring(
+				this.noteWidth,
+				{
+					toValue: Dimensions.get('window').width - this.state.notesBarWidth,
+					duration: 1500,
+				}
+			),
+		]).start();
+	};
+
+	private animateNotesBarClose = () => {
+		Animated.parallel([
+			Animated.spring(
+				this.notesBarPosition,
+				{
+					toValue: -1 * this.state.notesBarWidth,
+					duration: 1500,
+				}
+			),
+			Animated.spring(
+				this.notePosition,
+				{
+					toValue: -1 * this.state.notesBarWidth,
+					duration: 1500,
+				}
+			),
+			Animated.spring(
+				this.noteWidth,
+				{
+					toValue: Dimensions.get('window').width,
+					duration: 1500,
+				}
+			),
+		]).start();
+	};
+
+	private onNotesBarToggle = async () => {
+		if (this.props.showNotesBar === true) {
+			this.props.dispatch({ type: 'NOTES_BAR_CLOSE' });
+			this.animateNotesBarClose();
+		} else {
+			this.props.dispatch({ type: 'NOTES_BAR_OPEN' });
+			this.animateNotesBarOpen();
```

Comment:
It doesn't make sense that you have to both emit the action, and do execute the action (with animateBarOpen / Close). These animations should be a side effect of the action. Basically a component should watch for the state.showNotesBar and trigger the animation when it changes.

Also this has nothing to do in the Note component. Would probably be better in the parent component of the NotesBar

Reply:
It's not what I mean. What if the action NOTES_BAR_OPEN is dispatched from somewhere else? Well nothing would happen because opening/closing the bar is done here only. Any GUI action that needs to happen as a result of an action should be done as an effect, by using the state property as a dependency.

Reply:
The NOTES_BAR_OPEN action isn't for triggering the opening and closing of the notesbar, it seems the name is misleading. It should be something like 'IS_NOTESBAR_OPEN'. If 'IS_NOTESBAR_OPEN' is dispatched from somewhere else, the next time a note opens, it will do so with the notesbar open by default. That behaviour is necessary so that while switching between notes, you don't have to manually open the notesbar each time if it was open before, it will just stay opened. While opening the notesbar, I'm simply updating the global state with the action to show that the notesbar needs to stay open.

I could have added the state inside of root.tsx, but there's no way to pass props to the screens, so I made it a global state.

The effect that triggers the opening of the notesbar is pressing the toggle button.

Reply:
In any case all that shouldn't be in Note.tsx. The implementation is not right - the action should modify the state, and the components should then react and apply that state.

See for example NOTE_VISIBLE_PANES_SET, which modifies noteVisiblePanes, then components update themselves depending on that state variable. It allows dispatching NOTE_VISIBLE_PANES_SET from anywhere in the application and to know that the app will always be correctly updated.

If I dispatch NOTES_BAR_OPEN, the state will be "open" but the note bar won't visible because you need to manually trigger a bunch of animations.

Reply:
Alright, I'll work on fixing that.

Reply:
I just modified the code, the component now updates itself when the state changes.